### PR TITLE
MDS-2811 NoW status change push to NROS

### PR DESCRIPTION
**Pull request by JustinMacaulay**
Whenever a NoW status changes in core the change will be pushed to NROS.

### CODE REVIEW

File: services/core-api/app/api/now_applications/resources/now_application_import_resource.py [hunk 1]
```diff
@@ -65,5 +67,12 @@ def post(self, application_guid):
         if now_application_identity.now_application_id is not None:
             raise BadRequest('This record has already been imported.')
         application.save()
+        db.session.refresh(now_application_identity)
+        status = NOWApplicationStatus.find_by_now_application_status_code(
+            now_application_identity.now_application.now_application_status_code)
```
Comment (Jsyro) [inline services/core-api/app/api/now_applications/resources/now_application_import_resource.py:72]: now_application.now_application_status should return this. if it doesn't exist, can you add that relationship and use it here?

File: services/core-api/app/api/services/nros_now_status_service.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        resp = requests.post(
+            url=cls.nros_now_url,
+            headers={
+                'Authorization': f'Bearer {token}',
+                'Content-type': 'application/json',
+                'lobName': 'MMS'
+            },
+            data=json.dumps(data))
+        current_app.logger.info(f'Status for NoW Number{now_number} pushed to NROS')
+        return
```
Comment (Jsyro) [inline services/core-api/app/api/services/nros_now_status_service.py:45]: We should return something to the caller (even if they ignore it) and maybe log to the warning stream if the post responses.status_code is not 204, or something like that. If this needs to go live ASAP, we can defer this.

File: services/core-api/app/api/now_applications/resources/now_application_import_resource.py (hunk 1 above)
Comment (JustinMacaulay) [reply services/core-api/app/api/now_applications/resources/now_application_import_resource.py]: Added

File: services/core-api/app/api/services/nros_now_status_service.py (hunk 2 above)
Comment (JustinMacaulay) [reply services/core-api/app/api/services/nros_now_status_service.py:46]: Added more to the logger just to have this at least a bit more useful.

File: services/core-api/app/api/services/nros_now_status_service.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                'Content-type': 'application/json',
+                'lobName': 'MMS'
+            },
+            data=json.dumps(data))
+        current_app.logger.info(
+            f'Status for NoW Number{now_number} pushed to NROS with response: {resp.__dict__}')
+        return
+
+    @classmethod
+    def get_nros_now_token(cls):
```
Comment (exzizt) [inline services/core-api/app/api/services/nros_now_status_service.py:49]: I think this function could use some error handling, e.g., to ensure that the status code received is 200 and if not throw an appropriate error message.

File: services/core-api/app/api/services/nros_now_status_service.py [hunk 4]
```diff
@@ -41,17 +43,28 @@ def nros_now_status_update(cls, now_number, status, status_updated_date):
...
+            resp.raise_for_status()
+            resp_body = json.loads(resp.text)
 
-        resp_body = json.loads(resp.text)
-
-        token = resp_body["access_token"]
+            token = resp_body["access_token"]
+        except:
+            current_app.logger.info(f'Failed to get a token from NROS. Response: {resp.__dict__}')
+            pass
```
Comment (exzizt) [inline services/core-api/app/api/services/nros_now_status_service.py:68]: I think the `pass` can be removed.

File: migrations/sql/V2020.05.08.12.51__update_now_statuses.sql [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+VALUES
+	('REC', 'Received', 90, true, 'system-mds', 'system-mds'),
+    ('PAP', 'Pending Approval', 50, true, 'system-mds', 'system-mds'),
+	('REI', 'Rejected-Initial', 100, true, 'system-mds', 'system-mds'),
+	('PCO', 'Permit Closed', 40, true, 'system-mds', 'system-mds'),
+	('NPR', 'No Permit Required', 110, true, 'system-mds', 'system-mds'),
+	('RCO', 'Referral Complete', 120, true, 'system-mds', 'system-mds')
+ON CONFLICT DO NOTHING;
+
+-- Set all of the current NoW statuses to the new default status (Submitted)
```
[the remark this answers is not in the document]
Comment (JustinMacaulay) [reply migrations/sql/V2020.05.08.12.51__update_now_statuses.sql]: removed

File: services/core-api/app/api/services/nros_now_status_service.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    nros_now_token_pass = Config.NROS_NOW_CLIENT_SECRET
+    nros_now_token_user = Config.NROS_NOW_CLIENT_ID
+    nros_now_token_url = Config.NROS_NOW_TOKEN_URL
+
+    @classmethod
+    def nros_now_status_update(cls, now_number, status, status_updated_date):
+
+        token = cache.get(NROS_NOW_TOKEN)
+        if not token:
+            token = cls.get_nros_now_token()
```
[the remark this answers is not in the document]
Comment (JustinMacaulay) [reply services/core-api/app/api/services/nros_now_status_service.py:22]: Added more logging around around failing calls.

File: services/core-api/app/api/services/nros_now_status_service.py (hunk 4 above)
Comment (JustinMacaulay) [reply services/core-api/app/api/services/nros_now_status_service.py:68]: I dont think its needed but I put it in just to show it was a deliberate choice to ignore the error.

**Pull request merged**